packages/pyright-scip/src/lib.ts

```diff
@@ -40,19 +40,56 @@ export function formatSnapshot(
     const out: string[] = [];
     const symbolTable = getSymbolTable(doc);
 
+    const formatOptionsPrefix = '# format-options:';
+    const formatOptions = {
+        showDocs: false,
+        showRanges: false,
+    };
+
+    for (let line of input.lines) {
+        if (!line.startsWith(formatOptionsPrefix)) {
+            continue;
+        }
+
+        const options = line.slice(formatOptionsPrefix.length).trim().split(',');
+
+        for (let option of options) {
+            const optionName = option.trim();
+
+            if (!(optionName in formatOptions)) {
+                throw new Error(`Invalid format option: ${optionName}`);
+            }
+
+            formatOptions[optionName as keyof typeof formatOptions] = true;
+        }
+
+        break;
+    }
+
     const externalSymbolTable: Map<string, scip.SymbolInformation> = new Map();
     for (let externalSymbol of externalSymbols) {
         externalSymbolTable.set(externalSymbol.symbol, externalSymbol);
     }
 
+    const enclosingRanges: { range: Range; symbol: string }[] = [];
     const symbolsWithDefinitions: Set<string> = new Set();
+
     for (let occurrence of doc.occurrences) {
         const isDefinition = (occurrence.symbol_roles & scip.SymbolRole.Definition) > 0;
         if (isDefinition) {
             symbolsWithDefinitions.add(occurrence.symbol);
         }
+
+        if (occurrence.enclosing_range.length > 0) {
+            enclosingRanges.push({
+                range: Range.fromLsif(occurrence.enclosing_range),
+                symbol: occurrence.symbol,
+            });
+        }
```

Comment:
This data will be thrown away later if `!formatDocs.showRanges`. So instead of the early exit later on, it would be simpler to change this condition to be 
```suggestion
        if (formatOptions.showRanges && occurrence.enclosing_range.length > 0) {
            enclosingRanges.push({
                range: Range.fromLsif(occurrence.enclosing_range),
                symbol: occurrence.symbol,
            });
        }
```
and remove the 

```
        if (!formatOptions.showRanges) {
            return;
        }
```

later in the code.